packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -371,7 +422,12 @@ function hasCorrectNodeType<T extends Node>(
     return true;
 }
 
-function isMatchingElement(vnode: VBaseElement, elm: Element, renderer: RendererAPI) {
+function isMatchingElement(
+    vnode: VBaseElement,
+    elm: Element,
+    renderer: RendererAPI,
+    shouldValidateAttr: (attrName: string) => boolean = () => true
```

Comment:
Extracting out `(attrName: string) => boolean` as a `type` may make this look a bit cleaner